arch/posix/CMakeLists.txt

```diff
@@ -64,7 +64,15 @@ else() # Linux.x86_64
     check_set_compiler_property(APPEND PROPERTY fpsse2 "SHELL:-msse2 -mfpmath=sse")
     zephyr_compile_options($<TARGET_PROPERTY:compiler,fpsse2>)
     target_compile_options(native_simulator INTERFACE "$<TARGET_PROPERTY:compiler,fpsse2>")
-  endif ()
+
+    # Let's request from the host C library (glibc) that time_t be a 64bit type for x86 32bit builds
+    # so we avoid the year 2038 issue.
+    # https://sourceware.org/glibc/wiki/Y2038ProofnessDesign
+    if (CONFIG_ARCH_POSIX_FORCE_64BIT_TIME_T AND CONFIG_EXTERNAL_LIBC)
+      zephyr_compile_options(-D_TIME_BITS=64 -D_FILE_OFFSET_BITS=64)
```

Comment:
Maybe this is covered in the link provided, but why is `_FILE_OFFSET_BITS` also set?

Reply:
It is a constraint/requirement from glibc:
https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#index-_005fTIME_005fBITS
`_TIME_BITS=64 can be defined only when _FILE_OFFSET_BITS=64 is also defined.`

Reply:
Thanks for clarifying. I think `off_t` size would also just be fantastic to standardize as 64-bits for compatibility everywhere too. But that's a separate issue.